app/actions/views/root.js

```diff
@@ -60,7 +60,7 @@ export function goToNotification(notification) {
         if (channelId !== currentChannelId) {
             handleSelectChannel(channelId)(dispatch, getState);
         }
-        markChannelAsRead(teamId, channelId)(dispatch, getState).then(() => true).catch(() => true);
+        markChannelAsRead(channelId, currentChannelId)(dispatch, getState);
```

Comment:
The function signature is `markChannelAsRead(channelId, prevChannelId)`. I also removed the code that silenced all errors from this call.